File: sources/net.sf.j2s.core/test/dev/j2s/java/net/URI.js
Clazz.declarePackage ("java.net");
Clazz.load (["java.net.URL"], "java.net.URI", ["swingjs.JSToolkit"], function () {
c$ = Clazz.declareType (java.net, "URI", java.net.URL);
Clazz.defineMethod (c$, "toURL", 
function () {
try {
return swingjs.JSToolkit.checkForJSData (this,  new java.net.URL (this.toString ()));
} catch (e) {
if (Clazz.exceptionOf (e, java.net.MalformedURLException)) {
return null;
} else {
throw e;
}
}
});
});

File: sources/net.sf.j2s.java.core/src/java/net/URI.java
package java.net;

/**
 * A very minimal implementation
 * 
 * @author Bob Hanson
 *
 */
public class URI extends URL {
	
	public URI(String str) throws MalformedURLException {
		super(str);
	}
	
	public URL toURL() {
		return this;
	}
	
}

